### PR TITLE
Added Select and Unselect operations to handle slices with strides != 1.

### LINKED ISSUE #514: assignment issue
Minimal repro:
```
import torch
import torch_xla
import torch_xla_py.utils as xu
import torch_xla_py.xla_model as xm
import pdb
xla_device = xm.xla_device()
d = torch.ones((10, 4)).to(xla_device)
d[:, 0::2] = 2
pdb.set_trace()
print(d)
```
expected:
```
tensor([[2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.]])
```
but got all ones.

### PR DESCRIPTION
**Pull request by dlibenzi**
Fixes #514

### CODE REVIEW

File: torch_xla/csrc/ops/tensor_data.cpp [hunk 1]
```diff
@@ -59,23 +59,27 @@ TensorData::TensorData(at::Tensor tensor, Device device)
...
+    xla::Literal value = GetTensorLiteral(tensor_, &shape(), &device_);
+    return ReturnOp(xla::ConstantLiteral(loctx->builder(), value), loctx);
   }
-  xla::Literal value = GetTensorLiteral(tensor_, &shape(), &device_);
-  return ReturnOp(xla::ConstantLiteral(loctx->builder(), value), loctx);
+  xla::ComputationClient::DataPtr data = TensorToXlaData(tensor_, device_);
+  return ReturnOp(loctx->GetParameter(data), loctx);
 }
 
 bool TensorData::ShouldMakeConstant(const at::Tensor& tensor) {
```
Comment (asuhan) [inline torch_xla/csrc/ops/tensor_data.cpp]: We've reversed the semantics of this method and the name stayed the same. Was this wrong before? If not, I'd consider a different name for it.

File: torch_xla/csrc/ops/unselect.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "torch_xla/csrc/ir.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+
+class Unselect : public Node {
```
Comment (asuhan) [inline torch_xla/csrc/ops/unselect.h]: This name doesn't tell me much until I read the code. `GetSelectBase` would work better imo.

File: torch_xla/csrc/ops/tensor_data.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/ops/tensor_data.cpp]: Was wrong before. Used to be ShouldBeOnDevice(), but then I changed the name, but not reflected the implementation.

File: torch_xla/csrc/ops/unselect.h (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/csrc/ops/unselect.h]: Unselect as the inverse of Select. Like Undo is of Do :)
I'd leave it as is, unless we find a name which is clearly fitting better.

**Pull request merged**